packages/flutter/lib/src/painting/edge_insets.dart

```diff
@@ -822,32 +820,31 @@ class EdgeInsetsDirectional extends EdgeInsetsGeometry {
   /// [EdgeInsetsGeometry.lerp] static method.
   ///
   /// {@macro dart.ui.shadow.lerp}
-  static EdgeInsetsDirectional lerp(EdgeInsetsDirectional a, EdgeInsetsDirectional b, double t) {
+  static EdgeInsetsDirectional? lerp(EdgeInsetsDirectional? a, EdgeInsetsDirectional? b, double t) {
     assert(t != null);
     if (a == null && b == null)
       return null;
     if (a == null)
-      return b * t;
+      return b! * t;
     if (b == null)
       return a * (1.0 - t);
     return EdgeInsetsDirectional.fromSTEB(
-      ui.lerpDouble(a.start, b.start, t),
-      ui.lerpDouble(a.top, b.top, t),
-      ui.lerpDouble(a.end, b.end, t),
-      ui.lerpDouble(a.bottom, b.bottom, t),
+      ui.lerpDouble(a.start, b.start, t)!,
+      ui.lerpDouble(a.top, b.top, t)!,
+      ui.lerpDouble(a.end, b.end, t)!,
+      ui.lerpDouble(a.bottom, b.bottom, t)!,
     );
   }
 
   @override
-  EdgeInsets resolve(TextDirection direction) {
+  EdgeInsets resolve(TextDirection? direction) {
```

Comment:
Ditto here - if it's all because of the super class we should look at just making the super class non-nullable as well.